services/order.py

```diff
@@ -0,0 +1,37 @@
+from datetime import datetime
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+from db.models import Ticket, Order, MovieSession
+
+
+@transaction.atomic
+def create_order(
+    tickets: list[dict],
+    username: str,
+    date: datetime = None,
+) -> None:
+
+    order = Order.objects.create(
+        user=get_user_model().objects.get(username=username)
+    )
+
+    for ticket in tickets:
+        if date:
+            order.created_at = date
+            order.save()

```

Comment:
Modifying the `created_at` field after the order is created can lead to unexpected behavior, especially since it is set with `auto_now_add=True`. Consider setting the `created_at` field directly during the creation of the `Order` object if a specific date is needed, or avoid modifying it after creation.